sale_require_po_doc/models/sale_order.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (C) 2021 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    customer_need_po = fields.Boolean(
+        related="partner_id.customer_need_po",
+        string="Customer Requires PO",
+        readonly=True,
+    )
+    sale_doc = fields.Text(
+        related="partner_id.sale_doc",
+        string="Sales Require Documentation",
+        readonly=True,
+    )
+    sale_document_option = fields.Selection(
+        [
+            ("no_need", "No documentation needed for this Sale"),
+            ("done", "Documentation required obtained and archived"),
+        ],
+        string="Sale Documentation",
+    )
+    sale_document_note = fields.Text(string="Sale Documentation Notes")
+
+    def action_confirm(self):
+        for order in self:
+            if order.customer_need_po and not order.client_order_ref:
+                raise ValidationError(
+                    _(
+                        "You can not confirm sale order without \
+                    Customer reference."
```

Comment:
![image](https://user-images.githubusercontent.com/4667326/118679666-b11c9400-b81b-11eb-898b-e31e76a20328.png)

The error message shown is: "You can not confirm sale order without                     Customer reference."